proto/substrait/parameterized_types.proto

```diff
@@ -27,6 +27,7 @@ message ParameterizedType {
     Type.Time time = 17;
     Type.IntervalYear interval_year = 19;
     Type.IntervalDay interval_day = 20;
+    Type.IntervalMonth interval_month = 36;
```

Comment:
I aligned with the existing names, but personally I think I'd prefer `IntervalMonthDayNano` or similar to be more descriptive - thoughts?

Reply:
`IntervalMonthDayNano` or some variant seems reasonable (and better) here.

I don't think it makes sense to align with the existing names. If anything I want
* IntervalYear -> IntervalYearToMonth
* IntervalDay -> IntervalDayToSecond

to better align with the other names in the file, but that's not a backwards compatible change.